test/select.spec.js

```diff
@@ -399,6 +399,23 @@ describe('ui-select tests', function() {
     expect(getMatchLabel(el)).toEqual('Samantha');
   });
 
+  it('should correctly render initial state with track by $index', function () {
+    
+    var el = compileTemplate(
+      '<ui-select ng-model="selection.selected"> \
+        <ui-select-match placeholder="Pick one...">{{$select.selected.name}}</ui-select-match> \
+        <ui-select-choices repeat="person in people track by $index"> \
+          {{person.email}} \
+        </ui-select-choices> \
+      </ui-select>'
+    );
+
+    openDropdown(el);
+
+    var generatedId = el.scope().$select.generatedId;
+    expect($(el).find('[id$="row-' + generatedId + '-0"]').length).toEqual(1);

```

Comment:
Ah, I see what you're doing here - I think it would be better to just use the whole id, for simplicity.
```js
$(el).find('#ui-select-choices-row-' + generatedId + '-0')
```